api_v3_protocol_parameters.go

```diff
@@ -25,17 +26,31 @@ type V3ProtocolParameters struct {
 
 func NewV3ProtocolParameters(opts ...options.Option[V3ProtocolParameters]) *V3ProtocolParameters {
 	var schedulerRate WorkScore = 100000
+	var totalSupply BaseToken = 1813620509061365
```

Comment:
This needs to be done differently. You are just calculating around with default parameters. This is a constructor that takes options and constructs the parameters, so any of these can be changed from the outside. If you want to have sanity checks they need to be done after `options.Apply`